parser.ts

```diff
@@ -348,6 +388,57 @@ export function traverseStmt(c : TreeCursor, s : string) : Stmt<SourceLocation>
   }
 }
 
+function traverseDestructure(c: TreeCursor, s: string):DestructureLHS<SourceLocation>[] {
+  const lhsargs : DestructureLHS<SourceLocation>[] = [];
+  var location = getSourceLocation(c, s);
+  var hasStarred = 0;
+
+  do{
+    if(c.name === "AssignOp") 
+      break;
+    else if (c.type.name === ",") 
+      continue;
+    else {
+      var lhs = traverseDestructureLHS(c,s);
+      if(lhs.isStarred){
+        hasStarred = hasStarred + 1
+        if (hasStarred > 1)
+          throw new ParseError("Multiple starred expressions.", location.line)
+      }
+      lhsargs.push(lhs)
+    } 
+  } while(c.nextSibling())
+  // check if we want normal assignment expressions to have * or not
+
+  return lhsargs;
+}
+
+function traverseDestructureLHS(c: TreeCursor, s: string):DestructureLHS<SourceLocation> {
+  let isIgnore = false;
+  let isStarred = false;
+  // 1. check if star in from of variable name
+  if (c.name === "*") {
+    isStarred = true;
+    c.nextSibling();
+  }
+  const lhs = traverseAssignTarget(c,s);
+  // 3. check if ignore is encountered 
+  if (lhs.tag === "id" && lhs.name === "_"){
+    isIgnore = true;
+  }
+  return {lhs, isIgnore, isStarred};
+}
+
+function traverseAssignTarget(c: TreeCursor, s: string):AssignTarget<SourceLocation> {
+  var location = getSourceLocation(c, s);
+  const lhs = traverseExpr(c,s);
+  // 2. LHS is valid expression type : "id" | "lookup" 
+  if (lhs.tag!=="id" && lhs.tag!=="lookup") {
+    throw new ParseError("Cannot have "+ lhs.tag + " expression at LHS while parsing assignment statements.", location.line)
```

Comment:
Hmm seems like you are doing a bit of error reporting as well, again possibilities of conflicts. Please do talk to the error reporting team as well.

Reply:
We are working with the error reporting team to iron this out.